packages/flame/lib/src/effects2/controllers/standard_controller.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/animation.dart';
+
+import 'curved_effect_controller.dart';
+import 'delayed_effect_controller.dart';
+import 'effect_controller.dart';
+import 'infinite_effect_controller.dart';
+import 'linear_effect_controller.dart';
+import 'pause_effect_controller.dart';
+import 'repeated_effect_controller.dart';
+import 'reverse_curved_effect_controller.dart';
+import 'reverse_linear_effect_controller.dart';
+import 'sequence_effect_controller.dart';
+
+/// Factory function for producing common [EffectController]s.
+///
+/// In the simplest case, when only `duration` is provided, this will return
+/// a [LinearEffectController] that grows linearly from 0 to 1 over the period
+/// of that duration.
+///
+/// More generally, the produced effect controller allows to add a delay before
+/// the beginning of the animation, to animate both forward and in reverse,
+/// to iterate several times (or infinitely), to apply an arbitrary [Curve]
+/// making the effect progression non-linear, etc.
+///
+/// In the most general case, the animation proceeds through the following
+/// steps:
+///   1. wait for [startDelay] seconds,
+///   2. repeat the following steps [repeatCount] times (or [infinite]ly):
+///       a. progress from 0 to 1 over the [forwardDuration] seconds,
+///       b. wait for [atMaxDuration] seconds,
+///       c. progress from 1 to 0 over the [backwardDuration] seconds,
+///       d. wait for [atMinDuration] seconds.
+///
+/// If the animation is finite and there are no "backward" or "atMin" stages
+/// then the animation will complete at `progress == 1`, otherwise it will
+/// complete at `progress == 0`.
+EffectController standardController({
+  required double duration,
+  Curve? curve,
+  double reverseDuration = 0.0,
+  Curve? reverseCurve,
+  bool infinite = false,
+  int? repeatCount,
+  double startDelay = 0.0,
+  double atMaxDuration = 0.0,
+  double atMinDuration = 0.0,
+}) {
```

Comment:
Nice! I was scared at first that the user would have a super verbose syntax when adding an effect controller with several of these options, but this looks good.

Reply:
And it obviates the need for `SimpleEffectController` too -- the returned controller will be as simple as needed to accommodate all the options.